contest/views.py

```diff
@@ -164,6 +171,30 @@ def ask(request):
                 form = ClarificationForm(request.POST)
                 if form.is_valid():
                     new_clarification = form.save()
-                    logger.info('Clarification: User %s create Clarification %s!' % (request.user.username, new_clarification.id))
+                    logger.info('Clarification: User %s create Clarification %s!' 
+                        % (request.user.username, new_clarification.id))
+                return redirect('contest:contest',contest)
+    return redirect('contest:archive')
+
+def reply(request):
+    try:
+        clarification = request.POST['clarification']
+        instance = Clarification.objects.get(pk = clarification)
+        contest_obj = instance.contest
+        contest = contest_obj.id
+    except:
+        logger.warning('Clarification: Can not reply Clarification!')
+        return redirect('contest:archive')
+    
+    if request.user.is_authenticated():
```

Comment:
it seems that the annotation can't put inside the function. Can only put right before the beginning of the function